armi/reactor/blocks.py

```diff
@@ -157,6 +158,47 @@ def __deepcopy__(self, memo):
 
         return b
 
+    def createBlankCopy(self):
```

Comment:
This might be better as a function somewhere else because this adds design details into the generic block class. This could be closer to where it's applied as a function for now and upgraded to a method later maybe.

Reply:
This implementation did become specific to a `HexBlock`, but it's maybe a functionality that should exist for other types of blocks also.

Reply:
Yeah, we will need to find a way to fix this.  I don't like the idea that we are forcing the uniform mesh to only work for hex assemblies now.  (Because this is used pretty high-level in `makeAssemWithUniformMesh()`).

Reply:
It still works with non-hex assemblies, it just uses `copy.deepcopy(block)`. It's the same behavior that existed before, which is less efficient but still works.